### PR TITLE
Update mitmproxy

### PR DESCRIPTION
**Pull request by yan12125**
#### Description

Update mitmproxy and add new dependencies

The new pyperclip is necessary. See https://github.com/mitmproxy/mitmproxy/pull/2898

###### Type(s)
- [ ] bugfix
- [ ] enhancement
- [ ] security fix
- [x] update

###### Tested on
macOS 10.13.3 17D102
Xcode 9.2 9C40b

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?
- [x] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: python/py-h11/Portfile [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+    test.run                yes
+    test.cmd                py.test-${python.branch}
+    test.target             h11
+
+    livecheck.type  none
+} else {
+    livecheck.type  regex
+    livecheck.url   https://pypi.python.org/pypi/${python.rootname}/json
+    livecheck.regex "\"${python.rootname}-(\[.\\d\]+)\\${extract.suffix}\""
```
Comment (mojca) [inline python/py-h11/Portfile]: I wonder what goes wrong with default values. I'm sure there must be something that should make these regexps less painful than this. There's `livecheck.type pypi` etc., but I would need to look it up in other ports.

Comment (yan12125) [reply python/py-h11/Portfile]: Ah, I just copied these lines from another port without thinking again :)

Yes livecheck.type pypi is a better approach. Changed all ports involved in this PR.

**Comment by pmetzger** [comment]
LGTM. @mojca ?

**Comment by mojca** [comment]
No need to wait for me. My comment was addressed.

**Comment by pmetzger** [comment]
Merged. Thank you for contributing to MacPorts!

**Pull request merged**